src/sage/arith/misc.py

```diff
@@ -2045,18 +2069,34 @@ def xgcd(a, b):
         sage: xgcd(y^2, a*h*y + b)
         (1, 7*a^2/b^2, (((-h)*a)/b^2)*y + 1/b)
     """
-    try:
+    if b is not None:
+        # xgcd of two elements
+        try:
+            return a.xgcd(b)
+        except AttributeError:
+            a = py_scalar_to_element(a)
+            b = py_scalar_to_element(b)
+        except TypeError:
+            b = py_scalar_to_element(b)
         return a.xgcd(b)
-    except AttributeError:
-        a = py_scalar_to_element(a)
-        b = py_scalar_to_element(b)
-    except TypeError:
-        b = py_scalar_to_element(b)
-    return a.xgcd(b)
+    else:
+        # xgcd of >=2 elements
+        if not isinstance(a, (tuple, list)):
+            raise TypeError("input `a` should be a tuple or a list")
+        if len(a) < 2:
+            raise ValueError("at least two elements should be given")
```

Comment:
Is there any reason not to allow the xgcd of a single element? Of course it would probably just be `a = 1×a` but it's maybe convenient in general algorithms.

Note that `gcd` allows that.

Reply:
Actually, gcd also allows an empty list. Not sure if that's reasonable here.

Reply:
I didn't know that `gcd` allows single element, that's why I only allowed two or more elements. I'd update it too.

For empty list, it gives integer zero, so I can just add as returning `(0,)`, but I also don't know if this is the best or just allow at least one elements.

Reply:
Of course I can add it for completeness, which looks good too.

Reply:
`(0,)` sounds good to me. I mean, nobody is going to type this in the SageMath repl, but it might show up in an algorithm and it's a reasonable answer.